### PR TITLE
JAMES-2566 Reprocessing - missing keys - multiple repositories sharing a same path

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java [hunk 1]
```diff
@@ -1044,6 +1046,36 @@ public void reprocessingAllTaskShouldIncludeDetails() throws Exception {
             .body("completedDate", is(notNullValue()));
     }
 
+    @Test
+    public void reprocessingAllTaskShouldNotFailWhenSeveralRepositoryWithSamePath() throws Exception {
```
Comment (rouazana) [inline server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java]: s/Repository/Repositories

File: server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java [hunk 2]
```diff
@@ -1072,6 +1104,36 @@ public void reprocessingAllTaskShouldClearMailRepository() throws Exception {
         assertThat(mailRepository.list()).isEmpty();
     }
 
+    @Test
+    public void reprocessingAllTaskShouldClearBothMailRepositoryWhenSamePath() throws Exception {
```
Comment (rouazana) [inline server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java]: idem

File: server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java [hunk 3]
```diff
@@ -1098,6 +1160,33 @@ public void reprocessingAllTaskShouldEnqueueMailsOnDefaultQueue() throws Excepti
             .containsOnly(NAME_1, NAME_2);
     }
 
+    @Test
+    public void reprocessingAllTaskShouldEnqueueMailsOfBothRepositoryOnDefaultQueueWhenSamePath() throws Exception {
```
Comment (rouazana) [inline server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java]: idem

File: server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/service/ReprocessingServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    private static final Optional<String> NO_TARGET_PROCESSOR = Optional.empty();
+
+    private ReprocessingService reprocessingService;
+    private MemoryMailRepositoryStore mailRepositoryStore;
+    private MailQueueFactory<ManageableMailQueue> queueFactory;
+    private FakeMail mail1;
+    private FakeMail mail2;
+    private FakeMail mail3;
+
+    @Before
```
Comment (trantienduchn) [inline server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/service/ReprocessingServiceTest.java]: junit 5?

Comment (chibenwa) [reply server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/service/ReprocessingServiceTest.java]: I want to keep this bugfix changeset small and uncouple it from huge refactorings.

Comment (rouazana) [reply server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/service/ReprocessingServiceTest.java]: nice :) just for the record, this PR is already +469 −31. You could already have split it, handling multi  repo in a different PR than mono repo issues.

Comment (chibenwa) [reply server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/service/ReprocessingServiceTest.java]: But lead to very similar fix, hence my choice.

IMO it did not pass the "show-stopper point"

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**